sdk/attestation/Azure.Security.Attestation/src/Models/PolicyCertificatesResult.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.Security.Attestation.Models
+{
+    [JsonConverter(typeof(PolicyCertificatesResultConverter))]
+    public partial class PolicyCertificatesResult
+    {
+        private IReadOnlyList<X509Certificate2> _certificateList;
+        private object _statelock = new object();
+
+        /// <summary>
+        /// Creates a new instance of a <see cref="PolicyCertificatesResult"/> object.
+        /// </summary>
+        public PolicyCertificatesResult()
+        {
+        }
+
+        /// <summary>
+        /// Returns the list of policy management certificates for this attestation instance.
+        /// </summary>
+        /// <returns>A list of <see cref="X509Certificate2"/> certificates which are used to sign incoming requests.</returns>
+        public IReadOnlyList<X509Certificate2> GetPolicyCertificates()
+        {
+            lock (_statelock)
+            {
+                if (_certificateList == null)
+                {
+                    List<X509Certificate2> certificates = new List<X509Certificate2>();
+                    foreach (var key in InternalPolicyCertificates.Keys)
+                    {
+                        if (key.X5C == null)
+                        {
+                            // the key returned must have a X5c property.
```

Comment:
Should we include this in the message of the exception?

Reply:
I think that would absolutely make sense, but I was worried about localization costs associated with adding text. I can add it in english but this should properly be localized.